html/pfappserver/root/src/views/Configuration/_store/preferences.js

```diff
@@ -0,0 +1,126 @@
+import { customRef } from '@vue/composition-api'
+import { defineStore } from 'pinia'
+import store from '@/store' // required for 'system/version'
+import apiCall from '@/utils/api'
+
+export const IDENTIFIER_PREFIX = 'pfappserver::' // transparently prefix all identifiers - avoid key collisions
+
+export const api = {
+  getPreference: id => {
+    return apiCall.getQuiet(`preference/${IDENTIFIER_PREFIX}${id}`).then(response => {
+      const { data } = JSON.parse(response.data.item.value)
+      return data
+    })
+  },
+  setPreference: (id, data) => {
+    if (data) {
+      let body = {
+        id: `${IDENTIFIER_PREFIX}${id}`,
+        value: JSON.stringify({
+          data,
+          meta: {
+            created_at: (new Date()).getTime(),
+            updated_at: (new Date()).getTime(),
+            version: store.getters['system/version']
+          }
+        })
+      }
+      return apiCall.getQuiet(['preference', `${IDENTIFIER_PREFIX}${id}`]).then(response => { // exists
+        const { data: { item: { value = null } = {} } = {} } = response
+        if (value) {
+          const { meta: { created_at = null } = {} } = JSON.parse(value)
+          if (created_at) { // retain `created_at`
+            body = {
+              id: `${IDENTIFIER_PREFIX}${id}`,
+              value: JSON.stringify({
+                data,
+                meta: {
+                  created_at: created_at,
+                  updated_at: (new Date()).getTime(),
+                  version: store.getters['system/version']
+                }
+              })
+            }
+          }
+        }
+        return apiCall.putQuiet(['preference', `${IDENTIFIER_PREFIX}${id}`], body).then(response => {
+          return response.data
+        })
+      }).catch(() => { // not exists
+        return apiCall.putQuiet(['preference', `${IDENTIFIER_PREFIX}${id}`], body).then(response => {
+          return response.data
+        })
+      })
+    }
+    else {
+      return apiCall.deleteQuiet(['preference', `${IDENTIFIER_PREFIX}${id}`]).then(response => {
+        return response
+      })
+    }
+  },
+  removePreference: id => {
+    return apiCall.deleteQuiet(['preference', `${IDENTIFIER_PREFIX}${id}`]).then(response => {
+      return response
+    })
+  }
+}
+
+export const useStore = defineStore({
```

Comment:
Do we need to destroy the store on logout?

Reply:
Store purged in https://github.com/inverse-inc/packetfence/pull/6397/commits/a866fdfd9593acf0bcdc719e9466ae72170f1152